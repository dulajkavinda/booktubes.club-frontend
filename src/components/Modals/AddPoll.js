import React, { useState } from "react";
import { Modal, Box, TextField, Button } from "@mui/material";

import { DateTimePicker } from "@mui/lab";
import AdapterDateFns from "@mui/lab/AdapterDateFns";
import LocalizationProvider from "@mui/lab/LocalizationProvider";

const style = {
  position: "absolute",
  top: "50%",
  left: "50%",
  transform: "translate(-50%, -50%)",
  width: 400,
  bgcolor: "background.paper",
  border: "2px solid #000",
  boxShadow: 24,
  p: 4,
};
export default function AddPoll({ open, setOpen }) {
  const handleClose = () => {
    setOpen(false);
  };

  const [datePickerValue, setDatePickerValue] = useState([]);
  const [pollname, setPollname] = useState([]);
  const [book1, setBook1] = useState([]);
  const [book2, setBook2] = useState([]);
  const [book3, setBook3] = useState([]);
  const [book4, setBook4] = useState([]);
  const [book5, setBook5] = useState([]);

  const handleCreatePoll = () => {
    console.log(datePickerValue);
    console.log(pollname);
    console.log(book1);
    console.log(book2);
    console.log(book3);
    console.log(book4);
    console.log(book5);
  };

  return (
    <Modal
      open={open}
      onClose={handleClose}
      aria-labelledby="modal-modal-title"
      aria-describedby="modal-modal-description"
    >
      <Box sx={style}>
        <TextField
          fullWidth
          label="Poll Name"
          id="pollname"
          value={pollname}
          onChange={(e) => setPollname(e.target.value)}
        />
        <Box height={10} />
        <TextField
          fullWidth
          label="Book 1"
          id="book1"
          value={book1}
          onChange={(e) => setBook1(e.target.value)}
        />
        <Box height={10} />
        <TextField
          fullWidth
          label="Book 2"
          id="book2"
          value={book2}
          onChange={(e) => setBook2(e.target.value)}
        />
        <Box height={10} />
        <TextField
          fullWidth
          label="Book 3"
          id="book3"
          value={book3}
          onChange={(e) => setBook3(e.target.value)}
        />
        <Box height={10} />
        <TextField
          fullWidth
          label="Book 4"
          id="book4"
          value={book4}
          onChange={(e) => setBook4(e.target.value)}
        />
        <Box height={10} />
        <TextField
          fullWidth
          label="Book 5"
          id="book5"
          value={book5}
          onChange={(e) => setBook5(e.target.value)}
        />
        <Box height={10} />
        <LocalizationProvider dateAdapter={AdapterDateFns} fullWidth>
          <DateTimePicker
            fullWidth
            renderInput={(props) => <TextField {...props} />}
            label="Poll Deadline"
            value={datePickerValue}
            minDateTime={new Date()}
            onChange={(newValue) => {
              setDatePickerValue(newValue);
            }}
          />
          <Box height={10} />
          <Button variant="contained" onClick={handleCreatePoll}>
            Create Poll
          </Button>
        </LocalizationProvider>
      </Box>
    </Modal>
  );
}
