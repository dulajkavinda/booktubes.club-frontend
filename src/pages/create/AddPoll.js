import {
  Modal,
  ModalOverlay,
  ModalContent,
  ModalHeader,
  ModalFooter,
  ModalBody,
  ModalCloseButton,
  useDisclosure,
} from "@chakra-ui/react";
import { Button, ButtonGroup } from "@chakra-ui/react";
import { Select } from "@chakra-ui/react";
export default function AddPoll({ isOpen, onClose }) {
  return (
    <>
      <Modal isOpen={isOpen} onClose={onClose}>
        <ModalOverlay />
        <ModalContent>
          <ModalHeader>Select Books</ModalHeader>
          <ModalCloseButton />
          <ModalBody>
            <Select placeholder="Select Book 1">
              <option value="option1">Option 1</option>
              <option value="option2">Option 2</option>
              <option value="option3">Option 3</option>
            </Select>
            <Select mt={2} placeholder="Select Book 2">
              <option value="option1">Option 1</option>
              <option value="option2">Option 2</option>
              <option value="option3">Option 3</option>
            </Select>
            <Select mt={2} placeholder="Select Book 3">
              <option value="option1">Option 1</option>
              <option value="option2">Option 2</option>
              <option value="option3">Option 3</option>
            </Select>
            <Select mt={2} placeholder="Select Book 4">
              <option value="option1">Option 1</option>
              <option value="option2">Option 2</option>
              <option value="option3">Option 3</option>
            </Select>
            <Select mt={2} placeholder="Select Book 5">
              <option value="option1">Option 1</option>
              <option value="option2">Option 2</option>
              <option value="option3">Option 3</option>
            </Select>
          </ModalBody>

          <ModalFooter>
            <Button mr={3} onClick={onClose}>
              Close
            </Button>
            <Button colorScheme="blue">Create Poll</Button>
          </ModalFooter>
        </ModalContent>
      </Modal>
    </>
  );
}
