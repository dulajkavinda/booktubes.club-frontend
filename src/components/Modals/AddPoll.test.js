import React from "react";
import { shallow, mount } from "enzyme";
import AddPoll from "./AddPoll";

let wrapper;
beforeEach(() => {
  wrapper = shallow(<AddPoll open={true} />);
});

describe("AddPoll basic tests", () => {
  it("is snapshot changed", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
