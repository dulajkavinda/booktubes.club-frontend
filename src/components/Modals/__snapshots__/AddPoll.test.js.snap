// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPoll basic tests is snapshot changed 1`] = `
<ForwardRef(Modal)
  aria-describedby="modal-modal-description"
  aria-labelledby="modal-modal-title"
  onClose={[Function]}
  open={true}
>
  <ForwardRef(Box)
    sx={
      Object {
        "bgcolor": "background.paper",
        "border": "2px solid #000",
        "boxShadow": 24,
        "left": "50%",
        "p": 4,
        "position": "absolute",
        "top": "50%",
        "transform": "translate(-50%, -50%)",
        "width": 400,
      }
    }
  >
    <ForwardRef(TextField)
      fullWidth={true}
      id="pollname"
      label="Poll Name"
      onChange={[Function]}
      value={Array []}
    />
    <ForwardRef(Box)
      height={10}
    />
    <ForwardRef(TextField)
      fullWidth={true}
      id="book1"
      label="Book 1"
      onChange={[Function]}
      value={Array []}
    />
    <ForwardRef(Box)
      height={10}
    />
    <ForwardRef(TextField)
      fullWidth={true}
      id="book2"
      label="Book 2"
      onChange={[Function]}
      value={Array []}
    />
    <ForwardRef(Box)
      height={10}
    />
    <ForwardRef(TextField)
      fullWidth={true}
      id="book3"
      label="Book 3"
      onChange={[Function]}
      value={Array []}
    />
    <ForwardRef(Box)
      height={10}
    />
    <ForwardRef(TextField)
      fullWidth={true}
      id="book4"
      label="Book 4"
      onChange={[Function]}
      value={Array []}
    />
    <ForwardRef(Box)
      height={10}
    />
    <ForwardRef(TextField)
      fullWidth={true}
      id="book5"
      label="Book 5"
      onChange={[Function]}
      value={Array []}
    />
    <ForwardRef(Box)
      height={10}
    />
    <LocalizationProvider
      dateAdapter={[Function]}
      fullWidth={true}
    >
      <ForwardRef(DateTimePicker)
        fullWidth={true}
        label="Poll Deadline"
        minDateTime={2022-03-20T13:24:54.918Z}
        onChange={[Function]}
        renderInput={[Function]}
        value={Array []}
      />
      <ForwardRef(Box)
        height={10}
      />
      <ForwardRef(Button)
        onClick={[Function]}
        variant="contained"
      >
        Create Poll
      </ForwardRef(Button)>
    </LocalizationProvider>
  </ForwardRef(Box)>
</ForwardRef(Modal)>
`;
